.clang-tidy

```diff
@@ -0,0 +1,21 @@
+---
+# TODO: We currently disable default checks with -*. While many checks provide little value or
+# produce false positives, try to incrementally enable most of them.
+# TODO: these checks are in progress of cleaning up
+Checks: >
+  -*,
+  bugprone-capturing-this-in-member-variable,
+  bugprone-move-forwarding-reference,
+  bugprone-return-const-ref-from-parameter,
+  bugprone-use-after-move,
+  clang-analyzer-cplusplus.PureVirtualCall,
```

Comment:
nit: We use these checks downstream, but clang static analyzer checks can be excruciatingly slow – might be something we can't afford to run automatically by the IDE

Reply:
Happy to remove them. I'm not sure which ones are extremely slow and which ones aren't. Can you recommend github code change?

Reply:
FWIW, the `clang-analyzer-*` ones, specifically, are extremely slow.

Reply:
Yup, dropping just those should be sufficient here (clang static analyzer can be useful, despite being really slow and having many false positives, we should find another place to enable them).